llvm/lib/Transforms/Vectorize/VPlanTransforms.h

```diff
@@ -114,6 +114,8 @@ struct VPlanTransforms {
   static void createInterleaveGroups(
       const SmallPtrSetImpl<const InterleaveGroup<Instruction> *> &InterleaveGroups,
       VPRecipeBuilder &RecipeBuilder, bool ScalarEpilogueAllowed);
+
+  static void unrollByUF(VPlan &Plan, unsigned UF, LLVMContext &Ctx);
```

Comment:
nit: better place unrollByUF() before optimizeForVFAndUF(), following their phase ordering, i.e., optimizeForVFAndUF() should appear last - can be sunk independently.